TypedDictType crash with mypy 1.12.0

# Bug report
Mypy 1.12.0 crashes on calls to Django's `.annotate()`. It looks like the cause is the addition of the `readonly_keys` arg to `TypedDictType` in [this PR](https://github.com/python/mypy/pull/17644).

## What's wrong
Repro:
```python
from django.db.models import QuerySet

from webserver.clients.models import Client


def annotate_client_queryset() -> QuerySet[Client]:
    return Client.objects.annotate(name="display_name")
```

Error:
```bash
$ mypy . --show-traceback
./webserver/client/public/query/client.py:10: error: INTERNAL ERROR -- Please try using mypy master on GitHub:
https://mypy.readthedocs.io/en/stable/common_issues.html#using-a-development-mypy-build
Please report a bug at https://github.com/python/mypy/issues
version: 1.12.0
Traceback (most recent call last):
  File "mypy/checkexpr.py", line 5847, in accept
  File "mypy/nodes.py", line 1969, in accept
  File "mypy/checkexpr.py", line 480, in visit_call_expr
  File "mypy/checkexpr.py", line 614, in visit_call_expr_inner
  File "mypy/checkexpr.py", line 1471, in check_call_expr_with_callee_type
  File "mypy/checkexpr.py", line 1565, in check_call
  File "mypy/checkexpr.py", line 1811, in check_callable_call
  File "mypy/checkexpr.py", line 1262, in apply_function_plugin
  File "/Users/ncvc/Library/Caches/pypoetry/virtualenvs/agent-webserver-p5YMsjuR-py3.12/lib/python3.12/site-packages/mypy_django_plugin/transformers/querysets.py", line 278, in extract_proper_type_queryset_annotate
    fields_dict = helpers.make_typeddict(api, fields=field_types, required_keys=set(field_types.keys()))
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ncvc/Library/Caches/pypoetry/virtualenvs/agent-webserver-p5YMsjuR-py3.12/lib/python3.12/site-packages/mypy_django_plugin/lib/helpers.py", line 405, in make_typeddict
    typed_dict_type = TypedDictType(fields, required_keys=required_keys, fallback=fallback_type)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: __init__() missing required argument 'readonly_keys' (pos 3)
./webserver/client/public/query/client.py:10: : note: use --pdb to drop into pdb
```

## How is that should be

No crash

## System information

- OS: Mac
- `python` version: 3.12
- `django` version: 5.1.2
- `mypy` version: 1.12.0
- `django-stubs` version: 5.1.0
- `django-stubs-ext` version: 5.1.0

**Crash Report**

Getting same error, admittedly, the code that crashes was wrong. As the Player model do not got a zone field. When i removed the code, mypy works.

`Ran mypy . --show-traceback`

**Traceback**

```
(backend) PS C:\Repos\untitledgame\backend> mypy . --show-traceback                     
.\game\repos\zone\zone.py:188: error: INTERNAL ERROR -- Please try using mypy master on GitHub:
https://mypy.readthedocs.io/en/stable/common_issues.html#using-a-development-mypy-build
Please report a bug at https://github.com/python/mypy/issues
version: 1.12.0
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "mypy\checkexpr.py", line 5847, in accept
  File "mypy\nodes.py", line 1969, in accept
  File "mypy\checkexpr.py", line 480, in visit_call_expr
  File "mypy\checkexpr.py", line 614, in visit_call_expr_inner
  File "mypy\checkexpr.py", line 1471, in check_call_expr_with_callee_type
  File "mypy\checkexpr.py", line 1565, in check_call
  File "mypy\checkexpr.py", line 1811, in check_callable_call
  File "mypy\checkexpr.py", line 1262, in apply_function_plugin
  File "C:\Users\andre\.virtualenvs\backend-jnn8gBG-\Lib\site-packages\mypy_django_plugin\transformers\querysets.py", line 352, in extract_proper_type_queryset_values
    row_type = helpers.make_typeddict(ctx.api, column_types, set(column_types.keys()))
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\andre\.virtualenvs\backend-jnn8gBG-\Lib\site-packages\mypy_django_plugin\lib\helpers.py", line 405, in make_typeddict
    typed_dict_type = TypedDictType(fields, required_keys=required_keys, fallback=fallback_type)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: __init__() missing required argument 'readonly_keys' (pos 3)
.\game\repos\zone\zone.py:188: : note: use --pdb to drop into pdb
```

**To Reproduce**

Code that fails

``` python
return [p["id"] for p in Player.objects.filter(zone__x=x, zone__y=y).values("id")]
```

Relevant model
``` python
class Player(Model):
   # ... do not have any zone Field
```

**Your Environment**

<!-- Include as many relevant details about the environment you experienced the bug in -->

- Mypy version used: 1.12.0
- Python version used: 3.12.1
- Operating system and version: Windows 11 Home

--- a/mypy_django_plugin/lib/helpers.py
+++ b/mypy_django_plugin/lib/helpers.py
@@ -396,13 +396,21 @@ def convert_any_to_type(typ: MypyType, referred_to_type: MypyType) -> MypyType:
 
 
 def make_typeddict(
-    api: Union[SemanticAnalyzer, CheckerPluginInterface], fields: Dict[str, MypyType], required_keys: Set[str]
+    api: Union[SemanticAnalyzer, CheckerPluginInterface],
+    fields: Dict[str, MypyType],
+    required_keys: Set[str],
+    readonly_keys: Set[str],
 ) -> TypedDictType:
     if isinstance(api, CheckerPluginInterface):
         fallback_type = api.named_generic_type("typing._TypedDict", [])
     else:
         fallback_type = api.named_type("typing._TypedDict", [])
-    typed_dict_type = TypedDictType(fields, required_keys=required_keys, fallback=fallback_type)
+    typed_dict_type = TypedDictType(
+        fields,
+        required_keys=required_keys,
+        readonly_keys=readonly_keys,
+        fallback=fallback_type,
+    )
     return typed_dict_type
 
 

--- a/mypy_django_plugin/transformers/models.py
+++ b/mypy_django_plugin/transformers/models.py
@@ -1183,6 +1183,7 @@ def get_annotated_type(
                     api,
                     fields={**annotations.items, **fields_dict.items},
                     required_keys={*annotations.required_keys, *fields_dict.required_keys},
+                    readonly_keys={*annotations.readonly_keys, *fields_dict.readonly_keys},
                 )
     else:
         annotated_model = helpers.lookup_fully_qualified_typeinfo(api, model_type.type.fullname + "@AnnotatedWith")

--- a/mypy_django_plugin/transformers/querysets.py
+++ b/mypy_django_plugin/transformers/querysets.py
@@ -275,7 +275,12 @@ def extract_proper_type_queryset_annotate(ctx: MethodContext, django_context: Dj
 
     fields_dict = None
     if field_types is not None:
-        fields_dict = helpers.make_typeddict(api, fields=field_types, required_keys=set(field_types.keys()))
+        fields_dict = helpers.make_typeddict(
+            api,
+            fields=field_types,
+            required_keys=set(field_types.keys()),
+            readonly_keys=set(),
+        )
 
     if fields_dict is not None:
         annotated_type = get_annotated_type(api, model_type, fields_dict=fields_dict)
@@ -349,5 +354,5 @@ def extract_proper_type_queryset_values(ctx: MethodContext, django_context: Djan
 
         column_types[field_lookup] = field_lookup_type
 
-    row_type = helpers.make_typeddict(ctx.api, column_types, set(column_types.keys()))
+    row_type = helpers.make_typeddict(ctx.api, column_types, set(column_types.keys()), set())
     return helpers.reparametrize_instance(default_return_type, [model_type, row_type])

--- a/requirements.txt
+++ b/requirements.txt
@@ -14,5 +14,5 @@ Django==5.1.2; python_version >= '3.10'
 -e .[redis,compatible-mypy,oracle]
 
 # Overrides:
-mypy==1.11.2
+mypy==1.12.0
 pyright==1.1.384

--- a/setup.py
+++ b/setup.py
@@ -32,7 +32,7 @@ dependencies = [
 
 # Keep compatible-mypy major.minor version pinned to what we use in CI (requirements.txt)
 extras_require = {
-    "compatible-mypy": ["mypy~=1.11.0"],
+    "compatible-mypy": ["mypy~=1.12.0"],
     "redis": ["redis"],
     "oracle": ["oracledb"],
 }

--- a/tests/typecheck/fields/test_base.yml
+++ b/tests/typecheck/fields/test_base.yml
@@ -171,7 +171,7 @@
             message = models.BinaryField()
         obj = EncodedMessage(b'\x010')
 
-        reveal_type(obj.message) # N: Revealed type is "Union[builtins.bytes, builtins.memoryview]"
+        reveal_type(obj.message) # N: Revealed type is "Union[builtins.bytes, builtins.memoryview[builtins.int]]"
 
 -  case: test_small_auto_field_class_presents_as_int
    main: |

